gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/DatasetTest.java

```diff
@@ -0,0 +1,226 @@
+    Iterator<BaseTableInfo> tableInfoIterator = tableInfoPage.values().iterator();
+    Iterator<Table> tableIterator = tablePage.values().iterator();
+    while (tableInfoIterator.hasNext() && tableIterator.hasNext()) {
+      assertEquals(tableInfoIterator.next(), tableIterator.next().info());
+    }
+    assertFalse(tableInfoIterator.hasNext());
+    assertFalse(tableIterator.hasNext());
+    assertEquals(tableInfoPage.nextPageCursor(), tablePage.nextPageCursor());
+    verify(bigqueryOptions);
+  }
+
+  @Test
+  public void testGet() throws Exception {
+    BaseTableInfo info = TableInfo.builder(TableId.of("dataset", "table1"), Schema.of()).build();
+    expect(bigquery.getTable(TableId.of("dataset", "table1"))).andReturn(info);
+    replay(bigquery);
+    Table table = dataset.get("table1");
+    assertNotNull(table);
+    assertEquals(info, table.info());
+  }
+
+  @Test
+  public void testGetNull() throws Exception {
+    expect(bigquery.getTable(TableId.of("dataset", "table1"))).andReturn(null);
+    replay(bigquery);
+    assertNull(dataset.get("table1"));
+  }
+
+  @Test
+  public void testCreateTable() throws Exception {
+    TableInfo info = TableInfo.builder(TableId.of("dataset", "table1"), Schema.of(FIELD)).build();
+    expect(bigquery.create(info)).andReturn(info);
+    replay(bigquery);
+    Table table = dataset.create("table1", Schema.of(FIELD));
+    assertEquals(info, table.info());
+  }
+
+  @Test
+  public void testCreateView() throws Exception {
+    ViewInfo info = ViewInfo.builder(TableId.of("dataset", "table2"), "QUERY").build();
+    expect(bigquery.create(info)).andReturn(info);
+    replay(bigquery);
+    Table table = dataset.create("table2", "QUERY");
+    assertEquals(info, table.info());
+  }
+
+  @Test
+  public void testCreateExternalTable() throws Exception {
+    ExternalTableInfo info = ExternalTableInfo.builder(TableId.of("dataset", "table3"),
+        ExternalDataConfiguration.of(ImmutableList.of("URI"), Schema.of(), FormatOptions.csv()))
+        .build();
+    expect(bigquery.create(info)).andReturn(info);
+    replay(bigquery);
+    Table table = dataset.create("table3", ExternalDataConfiguration.of(
+        ImmutableList.of("URI"), Schema.of(), FormatOptions.csv()));
+    assertEquals(info, table.info());
+  }
+
+  @Test
+  public void testLoad() throws Exception {
+    expect(bigquery.getDataset(DATASET_INFO.datasetId().dataset())).andReturn(DATASET_INFO);
+    replay(bigquery);
+    Dataset loadedDataset = Dataset.load(bigquery, DATASET_INFO.datasetId().dataset());
+    assertNotNull(loadedDataset);
+    assertEquals(DATASET_INFO, loadedDataset.info());
+  }
+
+  @Test
+  public void testLoadNull() throws Exception {
+    expect(bigquery.getDataset(DATASET_INFO.datasetId().dataset())).andReturn(null);
+    replay(bigquery);
+    assertNull(Dataset.load(bigquery, DATASET_INFO.datasetId().dataset()));
+  }
+
+  @Test
+  public void testLoadWithOptions() throws Exception {
```

Comment:
Actually, ignore this comment.  I missed that the `expect` checks that the options are specified.